distributor/lru.go

```diff
@@ -32,8 +32,12 @@ func (lru *cache) Put(key string, value interface{}) {
 	}
 	lru.mut.Lock()
 	defer lru.mut.Unlock()
-	if _, ok := lru.get(key); ok {
-		return
+	if v, ok := lru.get(key); ok {
+		if v == value {
+			return
+		} else {
+			lru.priority.Remove(lru.priority.Front())
```

Comment:
Ah, find code with `Next()` loop would be `O(n)` generally, but in this case, the element is supposed to be moved in the front. Thus, even if I use the loop code, most probably the loop can be finished in the first loop.
I will update this with writing the `Next()` and loop code to find element.